nearcore/src/lib.rs

```diff
@@ -46,32 +46,32 @@ pub fn store_path_exists<P: AsRef<Path>>(path: P) -> bool {
     fs::canonicalize(path).is_ok()
 }
 
-pub fn get_store_path(base_path: &Path) -> String {
+pub fn get_store_path(base_path: &Path) -> PathBuf {
     let mut store_path = base_path.to_owned();
     store_path.push(STORE_PATH);
     if store_path_exists(&store_path) {
         info!(target: "near", "Opening store database at {:?}", store_path);
     } else {
         info!(target: "near", "Did not find {:?} path, will be creating new store database", store_path);
     }
-    store_path.to_str().unwrap().to_owned()
+    store_path
 }
 
-pub fn get_default_home() -> String {
-    match std::env::var("NEAR_HOME") {
-        Ok(home) => home,
-        Err(_) => match dirs::home_dir() {
-            Some(mut home) => {
-                home.push(".near");
-                home.as_path().to_str().unwrap().to_string()
-            }
-            None => "".to_string(),
-        },
+pub fn get_default_home() -> PathBuf {
+    if let Ok(near_home) = std::env::var("NEAR_HOME") {
+        return near_home.into();
     }
+
+    if let Some(mut home) = dirs::home_dir() {
+        home.push(".near");
+        return home;
+    }
+
+    PathBuf::default()
 }
 
 /// Function checks current version of the database and applies migrations to the database.
-pub fn apply_store_migrations(path: &String, near_config: &NearConfig) {
+pub fn apply_store_migrations(path: &PathBuf, near_config: &NearConfig) {
```

Comment:
```suggestion
pub fn apply_store_migrations(path: &Path, near_config: &NearConfig) {
```

See https://github.com/rust-analyzer/rust-analyzer/blob/master/docs/dev/style.md#useless-types

Reply:
thanks for the link! fixed